lambdas/services/pds_api_service.py

```diff
@@ -1,23 +1,43 @@
 import json
+import logging
+import uuid
+from time import time
 
+import jwt
+import requests
+from botocore.exceptions import ClientError
 from models.pds_models import Patient, PatientDetails
-from requests.models import Response
-from utils.exceptions import (InvalidResourceIdException,
-                              PatientNotFoundException, PdsErrorException)
+from requests.models import Response, HTTPError
+from utils.exceptions import (
+    InvalidResourceIdException,
+    PatientNotFoundException,
+    PdsErrorException,
+)
 from utils.utilities import validate_id
 
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
 
-class PdsApiService:
-    def fetch_patient_details(self, nhs_number: str) -> PatientDetails:
-        validate_id(nhs_number)
 
-        response = self.fake_pds_request(nhs_number)
+class PdsApiService:
+    def __init__(self, ssm_service):
+        self.ssm_service = ssm_service
 
-        return self.handle_response(response, nhs_number)
+    def fetch_patient_details(
+        self,
+        nhs_number: str,
+    ) -> PatientDetails:
+        try:
+            validate_id(nhs_number)
+            response = self.pds_request(nhs_number, retry_on_expired=True)
+            return self.handle_response(response, nhs_number)
+        except ClientError as e:
+            logger.error(f"Error when getting ssm parameters {e}")
+            raise PdsErrorException("Failed to preform patient search")
 
     def handle_response(self, response: Response, nhs_number: str) -> PatientDetails:
         if response.status_code == 200:
-            patient = Patient.model_validate(response.content)
+            patient = Patient.model_validate(response.json())
```

Comment:
Might have missed the implementation when I wrote the mock service but is there a check for if the `model_validate `fails? Think it raises a `ValidationError`

Reply:
The lambda will catch ValidationError